### ISSUE TITLE
Support for other langauges when creating a "code" note

### ISSUE THREAD

**Issue by OnyxDelta** [zadam/trilium#192]
I love this application and its direction.  However, I for my workflow, I was saddened to see that OCaml and R are not supported as languages.  I was wondering if they could possibly be added?  I noticed that codemirror in the sources directory has support for R and "mllike" syntax highlighting, which would cover OCaml.

**Comment by zadam** [comment]
Hello, thanks for this report. You're right that CodeMirror supports both R and ML-like languages and many others. Actually the number of supported languages is the main problem here - listing all of them would make the list in UI uncomfortably long and that's why I made some selection which is supposed to cover most needs.

I'm experimenting with exposing the list as an API which would allow you to add new items, remove unnecessary ones, sort them in some specific order other than alphabetical etc.

Example of code to do that:
```javascript
const codeMimeTypes = api.getDefaultCodeMimeTypes();
codeMimeTypes.push({
    mime: "text/x-ocaml",
    title: "OCaml"
});

codeMimeTypes.push({
    mime: "text/x-rsrc",
    title: "R"
});

// sort alphabetically
codeMimeTypes.sort((a, b) => a.title > b.title ? 1 : -1);

api.setCodeMimeTypes(codeMimeTypes);
```
Such code note would have label run=frontendStartup so it would be run automatically. Would this work for you?

**Comment by OnyxDelta** [comment]
That would be absolutely amazing. Thank you so much for looking into this.

**Comment by zadam** [comment]
Released in 0.23.0. Documented here: https://github.com/zadam/trilium/wiki/Code-notes
